core/deliverservice/config.go

```diff
@@ -97,7 +97,9 @@ func LoadOverridesMap() (map[string]*orderers.Endpoint, error) {
 func (c *DeliverServiceConfig) loadDeliverServiceConfig() {
 	c.PeerTLSEnabled = viper.GetBool("peer.tls.enabled")
 
-	c.ReConnectBackoffThreshold = viper.GetFloat64("peer.deliveryclient.reConnectBackoffThreshold")
+	reconnectBackoffThresholdDuration := viper.GetDuration("peer.deliveryclient.reConnectBackoffThreshold")
+
+	c.ReConnectBackoffThreshold = reconnectBackoffThresholdDuration.Seconds()
```

Comment:
I'm not sure we need the Seconds() here. This converts it to seconds, no?

Reply:
Yes, `Seconds()` was used to convert duration to a float. This does mix up the units for the provided value vs default, do you have any concerns if we change to always process the value as `time.Duration`?

Reply:
**time.Duration** is preferable.

Reply:
🛠